pkg/cortex/modules.go

```diff
@@ -122,6 +122,10 @@ func (t *Cortex) initRing() (serv services.Service, err error) {
 
 func (t *Cortex) initRuntimeConfig() (services.Service, error) {
 	if t.Cfg.RuntimeConfig.LoadPath == "" {
+		if t.Cfg.LimitsConfig.PerTenantOverrideConfig == "" {
+			// no need to initialize module if load path is empty
+			return nil, nil
```

Comment:
I'm pretty sure leaving `t.RuntimeConfig` set to nil will cause problems elsewhere.

Reply:
Nitpick, but can we move this check *after* first check?

- First `if` -- check whether to use new or old-style options
- Second `if` -- if LoadPath is still empty, return quickly

The way it is done right now mixes these two concerns, that should be separate.

Reply:
Sure, thanks @pstibrany! Making the changes.